TestingHelpers/MockDirectoryInfo.cs

```diff
@@ -284,5 +329,10 @@ public override DirectoryInfoBase Root
                 return new MockDirectoryInfo(mockFileDataAccessor, mockFileDataAccessor.Directory.GetDirectoryRoot(FullName));
             }
         }
+
+        private void CheckDirectoryExists() 
+        {
+            if (MockFileData == null) throw new FileNotFoundException("File not found", directoryPath);
```

Comment:
Good point. I took a look at [the source](http://referencesource.microsoft.com/#mscorlib/system/io/filesysteminfo.cs,189) to see what DirectoryInfo does so I can mimic that.

It's using `DateTime.FromFileTime` and `DateTime.FromFileTimeUtc`. See the output from my system (Brisbane, AU - +10).

```
PS C:\Users\John> $a = [System.IO.DirectoryInfo]::new("c:\temperererer")
PS C:\Users\John> $a.LastWriteTime

Monday, 1 January 1601 10:00:00 AM


PS C:\Users\John> $a.LastWriteTimeUtc

Monday, 1 January 1601 12:00:00 AM


PS C:\Users\John> [System.DateTime]::FromFileTime(0)

Monday, 1 January 1601 10:00:00 AM


PS C:\Users\John> [System.DateTime]::FromFileTimeUtc(0)

Monday, 1 January 1601 12:00:00 AM


PS C:\Users\John>
```

I'll update those methods to return a default time.